src/re_arr.rs

```diff
@@ -221,6 +221,25 @@ impl<T, const N: usize> ReArr<T, N> {
         Self { arr, arr_len: N }
     }
 
+    // Create a new [`ReArr`] from an iterator reference, taking up to N items
+    // 
+    // Allows for initialization without consuming the iterator, leaving its
+    // remaining content for another procedure.
+    // 
+    // This is useful for ComboVec::from_iter, which needs to initialise both
+    // a ReArr and a Vec.
+    pub(crate) fn from_iter_ref(iter: &mut impl Iterator<Item = T>) -> Self {
+        let mut re_arr = Self::new();
+        for _ in 0..N {
+            if let Some(val) = iter.next() {
+                re_arr.push(val);
+                continue;
+            }
+            break;
```

Comment:
It's important to break out of the loop like this because non-standard/non-canonical iterators may return `Some` after the first `None`.